pkg/kube/krt/index.go

```diff
@@ -87,3 +114,102 @@ func toString(rk any) string {
 	}
 	return tk
 }
+
+type indexCollection[K comparable, O any] struct {
+	idx index[K, O]
+	id  collectionUID
+	// nolint: unused // (not true, its to implement an interface)
+	collectionName string
+}
+
+// nolint: unused // (not true, its to implement an interface)
+func (i indexCollection[K, O]) name() string {
+	return i.collectionName
+}
+
+// nolint: unused // (not true, its to implement an interface)
+func (i indexCollection[K, O]) uid() collectionUID {
+	return i.id
+}
+
+// nolint: unused // (not true, its to implement an interface)
+func (i indexCollection[K, O]) dump() CollectionDump {
```

Comment:
I think it would be nice to be able to include information here for debugging. Can probably add it later though.

Reply:
https://github.com/istio/istio/pull/55457

Reply:
Nice! Thanks